Phaethon/InternalApi/DataManagement/InvoiceManagement.cs

```diff
@@ -17,24 +17,108 @@ internal InvoiceManagement()
             _invoiceDa = new InvoiceDa();
         }
 
-        bool IInvoiceManagement.Create(Invoice invoice)
+        public bool CreateOrUpdate(Invoice invoice)
         {
-            return _invoiceDa.Create(invoice);
+            CompanyDa companyDa = new CompanyDa();
+            RepresentativeDa representativeDa = new RepresentativeDa();
+            ProductDa productDa = new ProductDa();
+            ItemDa itemDa = new ItemDa();
+            ElementDa elementDa = new ElementDa();
+
+            using (var db = new DatabaseContext())
+            {
+                using (var dbTransaction = db.Database.BeginTransaction())
+                {
+                    try
+                    {
+                        companyDa.CreateOrUpdate(db, invoice.Receiver.Company);
```

Comment:
why do you have to call same method with same object so many times? 
Can't you just call companyDa.CreateOrUpdate(invoice) and thats it?

Reply:
no, that it will not work for my usecase idk about urs

Reply:
what is urs?

Reply:
Yours